pkg/util/clusterdata/worker_profiles_task.go

```diff
@@ -74,12 +75,14 @@ func (ef *workerProfilesEnricherTask) FetchData(ctx context.Context, callbacks c
 			continue
 		}
 
-		machineProviderSpec, err := utilmachine.UnmarshalAzureProviderSpec(machineset.Name, utilmachine.MachineSet, machineset.Spec.Template.Spec.ProviderSpec.Value.Raw)
-
+		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(machineset.Spec.Template.Spec.ProviderSpec.Value.Raw, nil, nil)
 		if err != nil {
-			// If this happens, the azure machine provider spec type/apiversion may have been updated and
-			// we need to handle it appropriately
-			ef.log.Info(err.Error())
+			ef.log.Info(err)
+			continue
+		}
+		machineProviderSpec, ok := obj.(*machinev1beta1.AzureMachineProviderSpec)
+		if !ok {
+			ef.log.Infof("failed to read provider spec from the machine set %q: %T", machineset.Name, obj)
 			continue
 		}
 
```

Comment:
We should not be using continue here. The logic is backwards because it is not in a function, and should be. We should be using return. It'd be great to have this refactored instead of adding more things with `continue`

Reply:
@facchettos 

> We should not be using continue here. The logic is backwards because it is not in a function, and should be. We should be using return.

I don't understand your point. I read it as "If thre is an issue with one `MachineSet` - log and continue to the next one". So logic looks correct to me.

> It'd be great to have this refactored instead of adding more things with `continue`

Anyway we are reverting back to the code we had before we introduced `utilmachine.UnmarshalAzureProviderSpec`. See [this PR](https://github.com/Azure/ARO-RP/pull/2167/files#diff-daed99b70882a6c275d461b2973b19bbf1d09f34100f08ae95c4ce58f935df42L78-L82). We are not adding anything new here.

Refactoring of thing is out of scope of this PR. Feel free to submit a PR to refactor this.